### ISSUE TITLE
Changing from one forge module to another with the same name (but differnt author) is not detected

### ISSUE THREAD

**Issue by antevens** [puppetlabs/r10k#63]
When a Puppetfile is changed from using a module like say: sdarwin/nrpe to using a totally different module with the same module name like: example42/nrpe this is not detected and the module is not purged/installed (as it should be?).

```
[R10K::Task::Deployment::PurgeEnvironments - DEBUG] No stale environments in /etc/puppet/environments

[R10K::TaskRunner - ERROR] Task #<R10K::Task::Module::Sync:0x7fe5fa059708> failed while running: "puppet module --modulepath '/etc/puppet/environments/master/modules' upgrade --ignore-dependencies example42/nrpe" returned with non-zero exit value #<Process::Status: pid=14992,exited(1)>
```

**Comment by adrienthebo** [comment]
Yeah, that seems like a case that I didn't anticipate. There are a cluster of issues around switching versions of modules with the same names, I should be addressing them fairly soon. I'm going to tentatively target this at 1.2 since I'm trying to get 1.1 pushed out of the door Any Day Now (TM). Thanks for reporting this!

**Comment by adrienthebo** [comment]
This should be fixed in 1.2.0; please reopen this issue if this is still a problem.
